### PR TITLE
databases weeks homework

### PR DESCRIPTION
**Pull request by CharlesKimMaina**

### CODE REVIEW

File: databases/week2/homework2.txt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+ORDER BY month created 
+
+
+Part 4: Creating a database
+/*Your application must include at least one many-to-many relationship and any supporting tables (linking tables) that are needed. The entity relationship diagram must describe what tables you will need, the columns in these tables, which column is the primary key, and the relationships between tables.*/
+ 
+
+
+CREATE TABLE ´user´(
+User_ID ,INT,
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:149]: user_id int,

File: databases/week2/homework2.txt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+name varchar(255),
+email VARCHAR (255),
+Phone varchar(255),
+PRIMARY KEY (User_ID)
+);
+
+CREATE TABLE `status`(
+status_ID, INT,
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:157]: status_id int,

File: databases/week2/homework2.txt [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+name varchar(255),
+PRIMARY KEY (status_id)
+);
+
+CREATE TABLE `task` (
+Task_ID, INT,
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:164]: task_id int,

File: databases/week2/homework2.txt (hunk 1 above)
Comment (CharlesKimMaina) [reply databases/week2/homework2.txt:149]: okey babak,.,i am reviewing them now

File: databases/week2/homework2.txt [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+PART 2.
+SCHOOL DATABASE.
+/*Create a new database containing the following tables:
+•	Class: with the columns: id, name, begins (date), ends (date)
+•	Student: with the columns: id, name, email, phone, class_id (foreign key)*/
+
+
+CREATE DATABASE schoolreg;
+
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:55]: Here you could add:

`use schoolreg;`

But it is okay

File: databases/week2/homework2.txt [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+`name` VARCHAR (255) NOT NULL,
+`begin_date`  DATETIME,
+`ending_date` DATETIME
+);
+
+CREATE TABLE `STUDENT` (
+`student_id` INT(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY, 
+`name` VARCHAR (255) NOT NULL,
+`email` VARCHAR(255) NOT NULL,
+`Phone` NULL,
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:67]: Phone needs a type. Probably varchar(45)

File: databases/week2/homework2.txt [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+FOREIGN KEY `class_id` INT (10)
+);
+
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:70]: How would solve this:
If the student has more than one classes?

File: databases/week2/homework2.txt [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+--Create an index on the name column of the student table.
+ALTER TABLE student ADD INDEX idx_name (name);
+--Or 
+CREATE INDEX idx_name
+ON student (name);
+/*Adding  a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).*/
+ALTER TABLE class 
+ADD status 
+Values ENUM(‘not started’, ‘ongoing’, ‘finished’);
+
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:82]: When you add the status column what would happen to the existing rows? Which value would status get.

You need to define a default value for this.

File: databases/week2/homework2.txt [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+WHERE user_id = 11;
+SELECT title, status_id 
+FROM task
+WHERE id IN (2, 17, 26, 27, 32);
+
+SELECT title
+FROM task
+INNER JOIN user_task 
+ON task.id = user_task.task_id
+WHERE user_task.user_id(
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:116]: where user_task.user_id `IN` (..

File: databases/week2/homework2.txt [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+SELECT id 
+FROM user
+WHERE name = ‘DONALD DUCK’
+AND status.id =(SELECT id FROM status);
+JOIN task on user.id = task.id;
```
Comment (babakbandpey) [inline databases/week2/homework2.txt:121]: This part will not work.
The status table is not included in table names.

File: databases/week2/homework2.txt (hunk 5 above)
Comment (CharlesKimMaina) [reply databases/week2/homework2.txt:67]: `phone` VARCHAR (45)

**Pull request merged**